x-pack/platform/plugins/shared/dataset_quality/public/state_machines/dataset_quality_details_controller/state_machine.ts

```diff
@@ -707,6 +740,45 @@ export const createPureDatasetQualityDetailsControllerStateMachine = (
           fieldLimit: undefined,
         })),
         raiseForceTimeRangeRefresh: raise('UPDATE_TIME_RANGE'),
+        storeFailureStoreUpdateParams: assign((_, event) => {
+          return 'failureStoreEnabled' in event
+            ? {
+                failureStoreUpdate: {
+                  params: {
+                    dataStream: '',
+                    failureStoreEnabled: event.failureStoreEnabled,
+                    customRetentionPeriod: event.customRetentionPeriod,
+                  },
+                },
+              }
+            : {};
+        }),
+        storeFailureStoreUpdateResponse: assign(
+          (context, event: DoneInvokeEvent<UpdateFailureStoreResponse>) => {
+            return 'data' in event
+              ? {
+                  failureStoreUpdate: {
+                    ...context.failureStoreUpdate,
+                    result: event.data,
+                    error: false,
+                  },
+                }
+              : {};
+          }
+        ),
+        storeFailureStoreUpdateError: assign((context) => {
```

Comment:
Why are we saving this into the state?
By just notifying the user that the update went wrong is enough, we are not using the error subsequently in the UI